src/components/input/input-editable.jsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import injectSheet from 'react-jss';
+import classNames from 'classnames';
+
+import styles from './input-editable-styles';
+import { Button, Icon, Input, Spinner } from '../../';
+
+class InputEditable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editing: false,
+      saving: false,
+      value: this.props.value,
+      originalValue: '',
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.value) this.setState({ value: nextProps.value });
+  }
+
+  onEdit = () => {
+    this.setState({ originalValue: this.state.value, editing: !this.state.editing });
+  };
+
+  onFocus = event => {
+    if (this.props.onFocus) {
+      this.props.onFocus(event);
+    }
+  };
+
+  onBlur = event => {
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  };
+
+  onChange = event => {
+    this.setState({ value: event.target.value || this.props.emptyDefaultValue });
+
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+  };
+
+  onSubmit = async event => {
+    event.preventDefault();
+    if (!this.props.hasError) {
+      if (this.props.onSubmit) {
+        this.setState({ saving: true });
+        await this.props.onSubmit(this.state.value);
+        this.setState({ saving: false });
+      }
+      this.setState({ editing: !this.state.editing });
+    }
```

Comment:
Maybe safest to always set it to false here? I don't see any case where we want it to be true in this case (?)

Reply:
Good suggestion, will fix!